lib/modules/commentNavigator.js

```diff
@@ -241,13 +253,17 @@ function changeCategory() {
 	if (index === -1) {
 		return;
 	}
-	currentCategory = (commentNav().select.options.item(index).value: any);
-	if (currentCategory !== '') {
+	const newCategory = (commentNav().select.options.item(index).value: any);
+	if (newCategory !== currentCategory) {
+		resetNavigator(newCategory);
```

Comment:
`getPostsByCategory` must be called before `resetNavigator` (so `currentCategory must be set before then).